src/sparseml/pytorch/sparsification/quantization/quantize.py

```diff
@@ -214,6 +221,72 @@ def add_input_activation_quant_wrappers(module: Module) -> Module:
     return module
 
 
+def add_output_activation_observers(module: Module):
+    """
+    implementation of torch.quantization add_observers_ that only adds observers
+    according to attached quantization_scheme properties. the existing implementation
+    (1.9+) includes its own logic for propagating including overriding set qconfigs
+    for certain activations without the ability to disable this behavior
+
+    :param module: module to add output activation observers to
+    """
+    # adapted from torch/ao/quantization/quantize.py::_add_observer_
+    device = next(module.parameters()).device
+
+    def _needs_observer(target_module: Module):
```

Comment:
```suggestion
    def _needs_observer(target_module: Module):
        # NOTE: this is a difference from the original implementation, which just checked for qconfig
```

Reply:
they also check for the descendent stuff, just in a different way (ie through leaf module checks and recursion), will throw in a similar comment

Reply:
Yeah just noticed all that